community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/FullCheck.java

```diff
@@ -79,8 +81,20 @@ public ConsistencySummaryStatistics execute( DirectStoreAccess stores, StringLog
         DiffRecordAccess records = recordAccess( stores.nativeStores() );
         execute( stores, decorator, records, report );
         ownerCheck.scanForOrphanChains( progressFactory );
-        countsBuilder.checkCounts( stores.nativeStores().getCounts(), new ConsistencyReporter( records, report ),
-                progressFactory );
+        CountsAccessor counts = stores.nativeStores().getCounts();
+        if ( counts instanceof CountsTracker )
+        {
+            CountsTracker tracker = (CountsTracker) counts;
+            try
+            {
+                tracker.start();
+            }
+            catch ( Exception e )
+            {
+                // let's hope it was already started :)
+            }
+        }
```

Comment:
This is super ugly. Now that counts creation has been internalised into the counts store, this might be possible to simplify by renaming `NeoStore.rebuildCountStoreIfNeeded()` to `NeoStore.startCountsStore()` or better yet, find some `LifeSupport` instance that has the appropriate cycle for the `CountsTracer` to hook into. But that doesn't answer the question of where this method should be invoked in this case (the case of consistency checking), since that doesn't bring the rest of the stores to the same state of the life-cycle.

Possibly the appropriate solution is to have some way for "tooling", such as this use case, to circumvent the state machine, and just read the store-files.

I fear we might just have to remember to change this code later, at least for a little while.